io/io/src/TBufferMerger.cxx

```diff
@@ -49,6 +49,10 @@ TBufferMerger::~TBufferMerger()
 
    if (!fQueue.empty())
       Merge();
+
+   TFile *out = fMerger.GetOutputFile();
+   if (out)
+      out->Write("",TObject::kOverwrite);
```

Comment:
Added:
```
   // Since we support purely incremental merging, Merge does not write the target objects
   // that are attached to the file (TTree and histograms) and thus we need to write them
   // now.
```